File: pkg/java/src/main/java/dev/openfga/language/errors/SyntaxError.java
package dev.openfga.language.errors;

import org.antlr.v4.runtime.RecognitionException;

public class SyntaxError extends ParsingError {

    private Metadata metadata;
    private RecognitionException cause;

    public SyntaxError(ErrorProperties properties, Metadata metadata, RecognitionException cause) {
        super(ErrorType.SYNTAX.getValue(), properties);
        this.metadata = metadata;
        this.cause = cause;
    }

    public Metadata getMetadata() {
        return metadata;
    }

    public void setMetadata(Metadata metadata) {
        this.metadata = metadata;
    }

    public RecognitionException getCause() {
        return cause;
    }

    public void setCause(RecognitionException cause) {
        this.cause = cause;
    }
}

File: pkg/csharp/Errors/SyntaxError.cs
using Antlr4.Runtime;

namespace OpenFgaLanguage.Errors;

public class SyntaxError : ParsingError
{
    public Metadata? Metadata { get; set; }
    public RecognitionException? Cause { get; set; }

    public SyntaxError() { }

    public SyntaxError(ErrorProperties properties, Metadata? metadata, RecognitionException? cause) 
        : base("syntax", properties)
    {
        Metadata = metadata;
        Cause = cause;
    }
}
